pkg/merge/merge_impl/group_by/group_by_stream_merge_rows.go

```diff
@@ -0,0 +1,165 @@
+//
+// Licensed to Apache Software Foundation (ASF) under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Apache Software Foundation (ASF) licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+//
+
+package group_by
+
+import (
+	"container/heap"
+	"fmt"
+	"github.com/dubbogo/arana/pkg/merge"
+	"github.com/dubbogo/arana/pkg/merge/aggr_unit"
+	"github.com/dubbogo/arana/pkg/proto"
+	"github.com/dubbogo/arana/pkg/runtime/xxast"
+	"github.com/dubbogo/arana/testdata"
+	"github.com/golang/mock/gomock"
+	"sync"
+)
+
+type GroupByStreamMergeRows struct {
+	mu           sync.RWMutex
+	stmt         xxast.SelectStatement
+	queue        merge.PriorityQueue
+	currentRow   proto.Row
+	isFirstMerge bool
+}
+
+func NewGroupByStreamMergeRow(rows []*merge.MergeRows, stmt xxast.SelectStatement) *GroupByStreamMergeRows {
```

Comment:
You don't need to depend upon `SelectStatement`, use field name directly instead, because sometimes the select element may not be a column name. For example, `select age,count(*)+1 from tbl ... group by -age`, here the `group_by` is `-age` and the second select element is a math expression.